src/hotspot/share/prims/jvm.cpp

```diff
@@ -1497,20 +1497,52 @@ static void bounds_check(const constantPoolHandle& cp, jint index, TRAPS) {
   }
 }
 
-JVM_ENTRY(jobjectArray, JVM_GetMethodParameters(JNIEnv *env, jobject method))
+JVM_ENTRY(jobjectArray, JVM_GetMethodParameterNames(JNIEnv *env, jobject method))
 {
   // method is a handle to a java.lang.reflect.Method object
   Method* method_ptr = jvm_get_method_common(method);
   methodHandle mh (THREAD, method_ptr);
-  Handle reflected_method (THREAD, JNIHandles::resolve_non_null(method));
   const int num_params = mh->method_parameters_length();
 
   if (num_params < 0) {
     // A -1 return value from method_parameters_length means there is no
-    // parameter data.  Return null to indicate this to the reflection
-    // API.
+    // parameter data.
     assert(num_params == -1, "num_params should be -1 if it is less than zero");
     return (jobjectArray)nullptr;
+  } else {
+    // This is called after JVM_GetMethodParameterFlags, which has already done validation.
+    objArrayOop result_oop = oopFactory::new_objArray(vmClasses::String_klass(), num_params, CHECK_NULL);
+    objArrayHandle result (THREAD, result_oop);
+
+    for (int i = 0; i < num_params; i++) {
+      MethodParametersElement* params = mh->method_parameters_start();
+      // For a 0 index, give a null symbol
+      Symbol* sym = 0 != params[i].name_cp_index ?
+        mh->constants()->symbol_at(params[i].name_cp_index) : nullptr;
+      if (nullptr != sym) {
+        Handle name = java_lang_String::create_from_symbol(sym, CHECK_NULL);
```

Comment:
Style: Typically we perform comparisons with the constant on the RHS, `x == 0` not `0 == x `.